content/src/ports/activeEntities.ts

```diff
@@ -337,25 +329,26 @@ export function createActiveEntitiesComponent(
    * Retrieve active entities that are pointed by pointers that match the urn prefix
    */
   async function withPrefix(
-    database: DatabaseClient,
     collectionUrn: string,
     offset: number,
     limit: number
   ): Promise<{ total: number; entities: Entity[] }> {
+    const database = components.database
     const parsedUrn = await parseUrn(collectionUrn)
     const isThirdPartyCollection = parsedUrn?.type === 'blockchain-collection-third-party-collection'
-
     const entityIds = await (isThirdPartyCollection
       ? thirdPartyCollectionItemsEntityIdsByPrefixCache.fetch(collectionUrn)
       : collectionItemsEntityIdsByPrefixCache.fetch(collectionUrn))
 
-    if (!entityIds) {
-      throw new Error(`error fetching urns for collection: ${collectionUrn}`)
+    // If the entity ids are not found, return an empty list
+    if (!entityIds || entityIds.length === 0) {
+      return { total: 0, entities: [] }
     }
 
+    const paginatedEntityIds = entityIds.slice(offset, offset + limit)
     const entities = await (isThirdPartyCollection
-      ? findThirdPartyCollectionItemsEntities(entityIds)
-      : withIds(database, entityIds.slice(offset, offset + limit)))
+      ? findThirdPartyCollectionItemsEntities(paginatedEntityIds)
```

Comment:
Is this a breaking change since pagination wasn't supported for TPW before? I think we now support it because of the materialized view, and it won't impact the response because we were paginating at a higher level after this query but just double-checking